code/modules/mob/living/carbon/xenomorph/castes/widow/abilities_widow.dm

```diff
@@ -0,0 +1,291 @@
+
+	newspit.generate_bullet(web_spit, web_spit.damage * SPIT_UPGRADE_BONUS(X))
+	newspit.permutated += X
+	newspit.def_zone = X.get_limbzone_target()
+
+	newspit.fire_at(target, X, null, newspit.ammo.max_range)
+	succeed_activate()
+	add_cooldown()
+
+// ***************************************
+// *********** Leash Ball
+// ***************************************
+
+/datum/action/xeno_action/activable/leash_ball
+	name = "leash_ball" // change to in character name later
+	ability_name = "leash_ball" // change to in character name later
+	mechanics_text = " Spit a huge ball of web that snares groups of marines "
+	action_icon_state = "scatter_spit" // temporary until I get my own icons
+	plasma_cost = 1
+	cooldown_timer = 1 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_LEASH_BALL
+
+/datum/action/xeno_action/activable/leash_ball/use_ability(atom/A)
+	var/turf/target = get_turf(A)
+	var/mob/living/carbon/xenomorph/X = owner
+	X.face_atom(target)
+	if(!do_after(X, 1 SECONDS, TRUE, X, BUSY_ICON_DANGER)) // currently here for balance prediction, shooting a 5x5 AoE snare is pretty insane even for T3 imo
+		return fail_activate()
+
+	var/datum/ammo/xeno/acid/web/leash_ball = GLOB.ammo_list[/datum/ammo/xeno/acid/web/leash_ball]
+	var/obj/projectile/newspit = new /obj/projectile(get_turf(X))
+
+	newspit.generate_bullet(leash_ball)
+	newspit.fire_at(target, X, null, newspit.ammo.max_range)
+	succeed_activate()
+	add_cooldown()
+
+/obj/structure/xeno/aoe_leash
+	name = "Snaring Web"
+	icon = 'icons/obj/items/projectiles.dmi' // temp ?
+	icon_state = "boiler_gas2" // temp
+	desc = "Looks very sticky"
+	destroy_sound = "alien_resin_break"
+	obj_integrity = 0
+	max_integrity = 1920
+	/// Radius for how far the leash should affect humans and how far away they may walk
+	var/leash_radius = 5
+	/// How much more integrity aoe_leash gains per caught marine, it is preferable that max_integrity is this var * 8.
+	var/integrity_increase = 240
+	/// List of beams to be removed on obj_destruction
+	var/list/obj/effect/ebeam/beams
+	/// Beam var
+	var/obj/effect/ebeam/beam
+	layer = ABOVE_ALL_MOB_LAYER
+	anchored = TRUE
+
+/// Humans caught get beamed and registered for proc/check_dist, aoe_leash also gains increased integrity for each caught human
+/obj/structure/xeno/aoe_leash/Initialize(mapload)
+	. = ..()
+	var/list/obj/effect/ebeam/beams = list()
+	for(var/mob/living/carbon/human/victim in view(leash_radius, loc))
+		beam = (beam(victim, "beam_heavy", 'icons/obj/items/projectiles.dmi', INFINITY, INFINITY))
+		beams += beam
+		RegisterSignal(victim, COMSIG_MOVABLE_MOVED, .proc/check_dist)
+		obj_integrity = obj_integrity + integrity_increase
+		if(obj_integrity > max_integrity)
+			obj_integrity = max_integrity
+	if(!beams)
+		return INITIALIZE_HINT_QDEL
+
+/// To remove beams after the leash_ball is destroyed
+/obj/structure/xeno/aoe_leash/obj_destruction()
+	. = ..()
+	if(!beams)
+		return
+	for(beam in beams)
+		qdel(beam)
```

Comment:
Also potentially check that this is cleaned up on Destroy() without destruction (say by the beam registering signals or so)